src/Core/Services/RequestValidator.cs

```diff
@@ -369,6 +377,14 @@ public static void ValidatePrimaryKeyRouteAndQueryStringInURL(EntityActionOperat
                     continue;
                 }
 
+                if (upsertRequestCtx.FieldValuePairsInBody.ContainsKey(exposedName!) && column.Value.IsReadOnly)
+                {
+                    throw new DataApiBuilderException(
+                            message: $"Field '{exposedName}' provided in request body cannot be assigned a value.",
```

Comment:
No, the customer should not be including the field in the request. 
`null` is still a value? If it causes confusion, I can change the messaging.
Yes, it will be affected by the request-body-strict PR.  This PR goes in first and then the request-body-PR has to be accordingly accounted for.

Reply:
Yes because `null` can be used to change a column's value. 

The messaging should reflect how a user can fix the request. Perhaps mentioning that the column should not be included in this request.

Reply:
Updated.